src/prefect/cli/deploy.py

```diff
@@ -263,9 +264,20 @@ async def deploy(
             " will be populated from the flow's description."
         ),
     ),
+    version_type: str = typer.Option(
+        None,
+        "--version-type",
+        help="The type of version to use for this deployment.",
+    ),
     version: str = typer.Option(
         None, "--version", help="A version to give the deployment."
     ),
+    version_info: str = typer.Option(
```

Comment:
two cents: this feels like a complicated flag given that it accepts string JSON. If we want to expose this, I think it's best exposed in the `prefect.yaml` file as opposed to the CLI (for now at least).

Reply:
One option we have is just to not accept this parameter on the CLI, _only_ have `version_type: vcs:git` in `prefect.yaml`, and just don't support other types for now.  This was supposed to be a "steam valve" for any versioning type that we didn't know how to autodetect

Reply:
I think Docker image versioning would kind of require the user passing `--version-info` from the outside, but we could skip that for now.

Reply:
Interesting, I gotcha - yea I have a preference to not expose it _yet_ until the need justifies it